torch/_inductor/codegen/cpp.py

```diff
@@ -2609,11 +2611,14 @@ def reduction_combine_vec(
             raise NotImplementedError
 
     def indirect_assert(self, var, lower, upper, mask=None):
-        assert not mask, "do not support mask in indirect_indexing assertion"
         assert isinstance(var, CppCSEVariable)
         assert var.dtype is not None
         if not var.is_vec:
+            if isinstance(mask, CppCSEVariable) and mask.is_vec:
```

Comment:
is this equivalent to `if mask and mask.is_vec`, or can `mask` be of a different type?

Reply:
The signature is
```
def indirect_assert(
        self,
        var: Union[CSEVariable, str],
        lower: Optional[str],
        upper: Optional[str],
        mask: Optional[str] = None,
    )
```

in `torch._inductor.codegen.common.Kernel`

Should have been `Optional[CSEVariable, str]` just like `var`. I was guarding against `str`